snappiershot/compare.py

```diff
@@ -174,6 +179,15 @@ def _compare_sets(
             reason = f"Sets do not match. Missing: {missing}; Extra: {extra}"
             return self.differences.add(operations, reason)
 
+    def _compare_pandas(
```

Comment:
In the broad scheme of things, this seems inefficient. Right now we have:
```
<User's DataFrame> ---> <Encoded DataFrame> ---> <Snapshot File>
                                                        ↓
                                               <Encoded DataFrame>
                                                        ↓
                                                <User's DataFrame>
                                                        ↓
                                               <Encoded DataFrame>
                                                        ↓
<User's DataFrame> ---> <Encoded DataFrame> --> <Snapshot Assert>
```
when we could more simply have
```
<User's DataFrame> ---> <Encoded DataFrame> ---> <Snapshot File>
                                                        ↓
                                               <Encoded DataFrame>
                                                        ↓
<User's DataFrame> ---> <Encoded DataFrame> --> <Snapshot Assert>
```
or ideally (but it seems isn't possible)
```
<User's DataFrame> ---> <Encoded DataFrame> ---> <Snapshot File>
                                                        ↓
                                               <Encoded DataFrame>
                                                        ↓
                                                <User's DataFrame>
                                                        ↓
<User's DataFrame> ---------------------------> <Snapshot Assert>
```

So, is there any good reason why we shouldn't add pandas DataFrames so the default encoding step and always work with the encoded data frame object?